### ISSUE TITLE
Visualization Settings at Questions/Dashboard level don't get reset/removed if Question Source gets changed

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#49832]
### Describe the bug

This is very similar to what is happening in [49667](https://github.com/metabase/metabase/issues/49667)

What happens is that if you created a Model -> Then a Question based on this model (edited some Visualization Settings) -> Created a Dashboard (edited some Visualization Settings) and then you decide to change the Model like pointing it to a new Table all the Visualization Settings would remain pointed to the old table.

At metabase level the frontend is able to ignore it but the problem would end up popping out in Serialization. If you are using Serialization the way it's supposed to be used you shouldn't hit this issue cause the target and source environment should match but if you are duplicating stuff from one instance to another you might not be migrating the whole Data Model and some of these references would be lost and break serialization during import

Also you have no way of resetting these Viz settings since any other viz setting you apply on the new question would still keep referenced to the old model

### To Reproduce

In my case I used the App DB to show what is going on:

Connected to the App DB -> Created a Model form `core_users` -> Created a Question form this model (edited Viz zettings) -> Added this Question to a dashboard (edited Viz zettings)

Now go to the Question and swap the model with another model `api_key` and save the Question. You will notice that the result metadata for the question got updated but the Viz settings still points to the column in core_users `last_login` which is the column I applied some Viz Settings on (the is not present in the api_key table):

![Image](https://github.com/user-attachments/assets/42ef6110-db1c-4d5b-b5e5-3eade1397cb7)

Same thing for the Dashboard:

![Image](https://github.com/user-attachments/assets/eda5f5ee-7a83-4e22-a02a-798a7fc037ac)

### Expected behavior

There shouldn't be any invalid Viz Settings applied to Model/Source that isn't even in the Question. So I guess we need to find a way to drop those when the model/source is changed. 

On the serialization front until the above gets fixed we can add logic to check the references, and drop the invalid ones before serializing.

### Information about your Metabase installation

```JSON
Has been like this since forever I think ... I tested on 51
```

### Severity

Would be cosmetic at first but might be a problem for people using Serialization and are doing "strange" stuff ... Like losing that table/column reference

**Comment by npfitz** [comment]
I think this is actually expected. Unless something has changed, Viz Settings are always validated before rendering a particular visualization (this usually involves looking at the data that is returned by the query). If they fail validation, then we generate and use some defaults, which may get overwritten as the user interacts with the card. I'll let someone from the DashViz team confirm though.

**Comment by Tony-metabase** [comment]
I agree and that's what is happening cause the frontend is able to handle those bad references.

But these bad references end up breaking things like serialization. So if it's fine for the frontend it's not necessarily fine for the backend. We might need to rewrite the issue to be serialization specific in this case to handle these bad references. Note this is similar to the following:

https://github.com/metabase/metabase/issues/49667

But the root cause of evil here are bad references that aren't cleaned up

**Comment by Tony-metabase** [comment]
Checked this again and serialization also works (garbage in and garbage out) ... Se the Viz Settings still contain references to columns that don't exist same as the frontend but the flow doesn't break serialization.

So this can be closed ... @piranha also noticed some issue with "Loading" fields form the APP DB and will push a fix for that
